keras_cv/layers/aspp.py

```diff
@@ -0,0 +1,195 @@
+                to 0.0, which means no dropout is applied to the output.
+            **kwargs: Additional keyword arguments to be passed.
+        """
+        super().__init__(**kwargs)
+        self.level = level
+        self.dilation_rates = dilation_rates
+        self.num_channels = num_channels
+        self.activation = activation
+        self.dropout = dropout
+
+    def build(self, input_shape):
+        # Retrieve the input at the level so that we can get the exact shape.
+        if not isinstance(input_shape, dict):
+            raise ValueError(
+                f"Expect the inputs to be a dict with int keys, got {input_shape}"
+            )
+        input_shape_at_level = input_shape[self.level]
+
+        height = input_shape_at_level[1]
+        width = input_shape_at_level[2]
+        channels = input_shape_at_level[3]
+
+        # This is the parallel networks that process the input features with different
+        # dilation rates. The output from each channel will be merged together and feed
+        # to the output.
+        self.aspp_parallel_channels = []
+
+        # Channel1 with Conv2D and 1x1 kernel size.
+        conv_sequential = tf.keras.Sequential(
+            [
+                tf.keras.layers.Conv2D(
+                    filters=self.num_channels, kernel_size=(1, 1), use_bias=False
+                ),
+                tf.keras.layers.BatchNormalization(),
+                tf.keras.layers.Activation(self.activation),
+            ]
+        )
+        self.aspp_parallel_channels.append(conv_sequential)
+
+        # Channel 2 and afterwards are based on self.dilation_rates, and each of them
+        # will have conv2D with 3x3 kernel size.
+        for dilation_rate in self.dilation_rates:
+            conv_sequential = tf.keras.Sequential(
+                [
+                    tf.keras.layers.Conv2D(
+                        filters=self.num_channels,
+                        kernel_size=(3, 3),
+                        padding="same",
+                        dilation_rate=dilation_rate,
+                        use_bias=False,
+                    ),
+                    tf.keras.layers.BatchNormalization(),
+                    tf.keras.layers.Activation(self.activation),
+                ]
+            )
+            self.aspp_parallel_channels.append(conv_sequential)
+
+        # Last channel is the global average pooling with conv2D 1x1 kernel.
+        pool_sequential = tf.keras.Sequential(
+            [
+                tf.keras.layers.GlobalAveragePooling2D(),
+                tf.keras.layers.Reshape((1, 1, channels)),
+                tf.keras.layers.Conv2D(
+                    filters=self.num_channels, kernel_size=(1, 1), use_bias=False
+                ),
+                tf.keras.layers.BatchNormalization(),
+                tf.keras.layers.Activation(self.activation),
+                tf.keras.layers.experimental.preprocessing.Resizing(
```

Comment:
out of curiosity -- Isn't Resizing already core API?

Reply:
Done.